pkg/logs/requestor.go

```diff
@@ -103,6 +102,10 @@ func buildCmd(ctx context.Context, req logs.Request) *exec.Cmd {
 // the loop is based on the Decoder example in the docs
 // https://golang.org/pkg/encoding/json/#Decoder.Decode
 func streamLogs(ctx context.Context, cmd *exec.Cmd, out io.ReadCloser, msgs chan logs.Message) {
+	// without this sleep the channel seems to get stuck. This results in either no log messages
+	// being read by the Handler _or_ the messages are read but only flushed when the request
+	// timesout
+	time.Sleep(time.Millisecond)
```

Comment:
Is there a way to avoid this? My gut is telling me "I fixed an issue by adding a sleep" is always the wrong solution - wrong because timing bugs eventually break and in a really intermittent way.

Also OK to merge this and get the feature out to people, if we can spend a little time looking into a solution after merging?

Reply:
i will keep looking for what the underlying cause is, but i don't know what specific part is broken. If you have any suggestions, i am more than happy to remove this